src/FffGcodeWriter.cpp

```diff
@@ -143,83 +147,107 @@ void FffGcodeWriter::writeGCode(SliceDataStorage& storage, TimeKeeper& time_keep
-            }
-            mesh.layers[layer_nr].seam_vertex_index = PolygonUtils::findClosest(seam_pos, mesh.layers[layer_nr].parts[0].insets[0][0]).point_idx;
-            ++layer_nr;
-        }
+    storage.spiralize_wall_outlines.reserve(total_layers);
+    storage.spiralize_seam_vertex_indices.reserve(total_layers);
 
-        for (; layer_nr < mesh.layers.size(); layer_nr++)
-        {
-            SliceLayer& layer = mesh.layers[layer_nr];
-            const SliceLayer& last_layer = mesh.layers[layer_nr - 1];
+    int last_layer_nr = -1; // layer number of the last non-empty layer processed (for any extruder or mesh)
 
-            if (layer.parts.size() < 1 || layer.parts[0].insets.size() < 1 || last_layer.parts.size() < 1 || last_layer.parts[0].insets.size() < 1)
-            {
-                // either this layer or the last has no parts (or a part has no insets) so just duplicate the index calculated for the last layer
-                layer.seam_vertex_index = last_layer.seam_vertex_index;
-                continue;
-            }
-            ConstPolygonRef last_wall = last_layer.parts[0].insets[0][0];
-            ConstPolygonRef wall = layer.parts[0].insets[0][0];
-            const int n_points = wall.size();
-            Point last_wall_seam_vertex = last_wall[last_layer.seam_vertex_index];
-
-            // seam_vertex_idx is going to be the index of the seam vertex in the current wall polygon
-            // initially we choose the vertex that is closest to the seam vertex in the last spiralized layer processed
-
-            int seam_vertex_idx = PolygonUtils::findClosest(last_wall_seam_vertex, wall).point_idx;
-
-            // now we check that the vertex following the seam vertex is not to the right of the seam vertex in the last layer
-            // and if it is we move forward
-
-            // get the inward normal of the last layer seam vertex
-            Point last_wall_seam_vertex_inward_normal = PolygonUtils::getVertexInwardNormal(last_wall, last_layer.seam_vertex_index);
-
-            // create a vector from the normal so that we can then test the vertex following the candidate seam vertex to make sure it is on the correct side
-            Point last_wall_seam_vertex_vector = last_wall_seam_vertex + last_wall_seam_vertex_inward_normal;
+    for (unsigned layer_nr = 0; layer_nr < total_layers; ++layer_nr)
+    {
+        bool done_this_layer = false;
 
-            // now test the vertex following the candidate seam vertex and if it lies to the left or on the vector, it's good to use
-            const int first_seam_vertex_idx = seam_vertex_idx;
-            float a = LinearAlg2D::getAngleLeft(last_wall_seam_vertex_vector, last_wall_seam_vertex, wall[(seam_vertex_idx + 1) % n_points]);
+        // default is no information available
+        storage.spiralize_wall_outlines[layer_nr] = nullptr;
+        storage.spiralize_seam_vertex_indices[layer_nr] = 0;
```

Comment:
You could also have done `storage.spiralize_wall_outlines.resize(layer_count, nullptr);` instead of reserving and setting separately...

It's not required for you to change this, though.
Just saying...